caf-audit-monkey/README.md

```diff
@@ -0,0 +1,145 @@
+# Audit Monkey
+
+## Description
+The Audit Monkey provides the functional ability to test the Audit Service including the auto-generated AuditLog, (both the direct and webservice variants) and the Audit WebService.  
+The Audit Monkey has the ability to send Audit Events both directly to Elasticsearch and via the Audit WebService.
+The Audit Monkey also has the ability to generate significant volumes of data as it can be run in both single-threaded and multi-threaded modes.  
+
+## Configuration
+
+### Modes
+The Audit Monkey can be run in a number of modes. Details of these modes are outline in the table below.  
+The three key modes cover:  
+
+* Sending Audit Events directly to Elasticsearch or via the Audit WebService
+* Sending Audit Events continuously or in random mode featuring sleeps
+* Executing as a single-threaded process or as a multi-threaded process
+
+<table>
+  <tr>
+    <th>Mode</th>
+    <th>Options</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>CAF_AUDIT_MODE</td>
+    <td>direct, webservice</td>
+    <td>
+      <ul>
+        <li><b>Direct:</b> Audit Events are sent directly to Elasticsearch</li>
+        <li><b>WebService:</b> Audit Events are sent via the RESTful API of the Audit WebService to Elasticsearch</li>
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <td>CAF_AUDIT_MONKEY_MODE</td>
+    <td>standard, random</td>
+    <td>
+      <ul>
+        <li><b>Standard:</b> The Audit Monkey attempts to send the specified number of Audit Events as quickly as possible</li>
+        <li><b>Random:</b> The Audit Monkey attempts to send portions of the overall specified number of Audit Events interlaced with pauses of execution, to create a pseudo randam sequence of Audit Events</li>
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <td>CAF_AUDIT_MONKEY_NUM_OF_THREADS</td>
```

Comment:
Will depend on the resources available on the machine / VM.  Works with 20 threads, but at point thrashing occurs and performance degrades, not sure where exactly this point is.  5000 threads caused a catastrophic failure, found that by accident during testing.